system/jlib/jstream.cpp

```diff
@@ -896,8 +896,14 @@ class CBlockedSerialOutputStream final : public CInterfaceOf<IBufferedSerialOutp
     {
         if (buffer.length() < newLength)
         {
-            constexpr size32_t alignment = 32;
-            newLength = (newLength + (alignment - 1)) & ~(alignment -1);
+            //When serializing a child dataset with many rows (e.g. 500K) this could be called many times, each time
+            //causign the buffer to reallocate.  Therefore ensure the new size is rounded up to avoid excessive
+            //reallocation and copying.
+            size32_t alignment = blockWriteSize / 4;
```

Comment:
Possibly a better alternative is buffer.length() / 4.  Otherwise when the rows get very large (e.g. 100MB) there is still a very large amount of reallocation happening.
I will change.